src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2454,15 +2463,22 @@ void SessionImpl::banIP(const QString &ip)
         return;
 
     lt::error_code ec;
-    const lt::address addr = lt::make_address(ip.toLatin1().constData(), ec);
+    const std::optional<std::pair<QString, QString>> ip_range = Utils::Net::parseIpRange(ip);
+    if (!ip_range.has_value())
+        return;
+    const lt::address first = lt::make_address(ip_range.value().first.toLatin1().constData(), ec);
+    Q_ASSERT(!ec);
+    if (ec)
+        return;
+    const lt::address last = lt::make_address(ip_range.value().second.toLatin1().constData(), ec);
     Q_ASSERT(!ec);
     if (ec)
         return;
 
-    invokeAsync([session = m_nativeSession, addr]
+    invokeAsync([session = m_nativeSession, first, last]
     {
         lt::ip_filter filter = session->get_ip_filter();
-        filter.add_rule(addr, addr, lt::ip_filter::blocked);
+        filter.add_rule(first, last, lt::ip_filter::blocked);
```

Comment:
What's the point of this change? As far as I understand, it does not do anything, you are turning single IP to a range for no reason. Existing method should work just fine as is.

Reply:
Why? There is no such use case. This method is only called when user manually bans individual peers.

Reply:
> Why? There is no such use case. This method is only called when user manually bans individual peers.

Indeed, I revisited the code and initially thought this method would be called by the web API. However, I noticed it is only invoked by `TransferController::banPeersAction()`, which exclusively passes `addr.ip.toString()`. I will revert this change.